build-cargo-dep.sh

```diff
@@ -13,9 +13,17 @@ if tty -s; then
 	USE_TTY="-it"
 fi
 
-mkdir -p cargo-deps
 alias 'rust-arm64-builder'='docker run $USE_TTY --rm -v "$HOME/.cargo/registry":/root/.cargo/registry -v "$(pwd)"/cargo-deps:/home/rust/src -P start9/rust-arm-cross:aarch64'
+if [[ "$(uname -m)" =~ ^(arm64|aarch64)$ ]]; then
+	echo "Detected arm64 system, skipping cross-build"
+	cargo install "$1"
+	mkdir -p cargo-deps/aarch64-unknown-linux-gnu/release/
+	cp $(which "$1") cargo-deps/aarch64-unknown-linux-gnu/release/
+else
```

Comment:
check the *whole triple* please. Don't want this to break in a super non-obvious way if built on BSD or musl

Reply:
by "whole triple" do you mean:

```
   -m, --machine
          print the machine hardware name

   -p, --processor
          print the processor type or "unknown"

   -i, --hardware-platform
          print the hardware platform or "unknown"
```

If so, my ubuntu vm in UTM on macos gives:
```
cguida@cg-m1-ubuntu:~/embassy-os$ uname -m
aarch64
cguida@cg-m1-ubuntu:~/embassy-os$ uname -p
aarch64
cguida@cg-m1-ubuntu:~/embassy-os$ uname -i
aarch64
```

while my embassy gives:
```
root@embassy-30297ba3:~# uname -m
aarch64
root@embassy-30297ba3:~# uname -p
unknown
root@embassy-30297ba3:~# uname -i
unknown
```

and my mac gives:
```
cguida@cg-mac 22-08-01T14:18 embassy-os % uname -m
arm64
cguida@cg-mac 22-08-01T14:18 embassy-os % uname -p
arm
cguida@cg-mac 22-08-01T14:18 embassy-os % uname -i
uname: illegal option -- i
usage: uname [-amnprsv]
```

Do we currently support building on BSD and musl? What do you expect to break in those builds?

Reply:
I have no idea if it would even build, but the binary would be incorrect if so. 

The whole triple would be something like aarch64-linux-gnu. Arch, kernel, libc implementation.

Reply:
Do you have any suggestions as to how to check those things in bash?

Reply:
Looks like just `uname` gives `Linux` on the embassy and the ubuntu VM, whereas my mac gives `Darwin`. So that works.

Now to determine libc implementation...

Reply:
looks like `grep -q __GLIBC__ <<< "$(gcc -dM -E - <<< "#include <features.h>")"` is the shortest way to detect glibc